contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2677,7 +2677,11 @@ rewriteBatchLevelStatement(
 			{
 				auto options = cctx->function_option();
 				auto commas = cctx->COMMA();
-				GetTokenFunc<TSqlParser::Function_optionContext*> getToken = [](TSqlParser::Function_optionContext* o) { return o->SCHEMABINDING(); };
+				GetTokenFunc<TSqlParser::Function_optionContext*> getToken = [](TSqlParser::Function_optionContext* o) {
+					if (o->execute_as_clause())
+						return o->execute_as_clause()->CALLER();
+					return o->SCHEMABINDING();
+				};
```

Comment:
Many code blocks under `if (cctx->WITH())` are duplicate. Can we consider using a template function?